### ISSUE TITLE
Rule request: kebab-case scripts

### ISSUE THREAD

**Issue by Zamiell** [JoshuaKGoldberg/eslint-plugin-package-json#61]
It is conventional for scripts in the "package.json" file to be in kebab-case, as evidenced here:

- <https://github.com/angular/angular/blob/main/package.json>
- <https://github.com/facebook/react/blob/main/package.json>
- <https://github.com/microsoft/TypeScript/blob/main/package.json>

Thus, this plugin should warn when incorrectly named scripts exist such as "fooBar" (i.e. camelCase) or "FooBar" (i.e. PascalCase) or "foo_bar" (i.e. snake_case) or "FOO_BAR" (i.e. SCREAMING_SNAKE_CASE).

**Comment by JoshuaKGoldberg** [comment]
Makes sense! I can see value in having a naming conventions rule that enforces kebab-case. Other efforts have popped up to standardize them too:

* https://github.com/github/scripts-to-rule-them-all
* https://conventionalscripts.org (cc @bennycode)

I think for the first few iterations of this rule we probably wouldn't want it to be enabled by default - as it's rather pedantic and doesn't _directly_ catch runtime issues (only indirectly). But strong +1 to it existing.

Questions that I think would need to be resolved:

* What other conventions exist?
* What "preset" conventions should it allow?
* How should it be configured, in general?

**Comment by ic4l4s9c** [comment]
I'm interested in implementing this (as part of #903), so I'm curious to know what is the current stance on implementing naming conventions is.

In particular:
- Should enforcing naming convention for scripts be a part of #839?
- If so, should `kebab-case` be a default option?
- Any suggestions/recommendations on how should configuration look like?
  - Especially in case of implementing something like [Conventional Scripts](https://conventionalscripts.org/) naming convention

**Comment by michaelfaith** [comment]
I think it makes sense to have this be an opt-in configuration for `valid-scripts`, but I'm not sure that I'd recommend it being the default.  Since it's going to be part of the `recommended` config, I'd propose the rule's default behavior sticks to purely validating that the shape is correct (i.e. an object with literal string values).  I do like the idea of having an option to enforce conventional scripts.  It's always good to have a standard to point to like that.  What do you think @JoshuaKGoldberg?

**Comment by JoshuaKGoldberg** [comment]
Good questions!

> * Should enforcing naming convention for scripts be a part of [🚀 Feature: Bring in a valid-scripts rule #839](https://github.com/JoshuaKGoldberg/eslint-plugin-package-json/issues/839)?

For the sake of doing one thing at a time, my vote is leaving it as a followup. It's non-trivial and I'm sure we'll discover edge cases to be discussed in review.

> * If so, should `kebab-case` be a default option?

Agreed that it should not be. This isn't an industry standard at all _(yet?)_, so it would inconvenience a lot of users to have it enabled by default.

> * Any suggestions/recommendations on how should configuration look like?
>   
>   * Especially in case of implementing something like [Conventional Scripts](https://conventionalscripts.org/) naming convention

Ideating: maybe an option value like `"conventional:case" | "lowercase"`?

**Comment by michaelfaith** [comment]
Blocked by https://github.com/JoshuaKGoldberg/eslint-plugin-package-json/issues/839

**Comment by michaelfaith** [comment]
#839 was completed. This is unblocked.
